### PR TITLE
ruby 2 and rails 4

### PR DESCRIPTION
**Pull request by compwron**
Previous: https://github.com/grosser/ar_after_transaction/pull/10

1. Adding log to .gitignore because it came up in my local diffs
2. Removing old rubies from travis
3. Removing rails2 from Appraisal since it does not work with ruby2
4. Conditional gem requires in Appraisal for rails3 to work with ruby 2.2.2
5. Adding rack and railties and conditional require because railties moved to a separate gem for rails4
6. Syntax deprecation fixes in ar_after_transaction_spec and setup_database
7. Using new travis infrastructure as per http://docs.travis-ci.com/user/migrating-from-legacy/
8. Using new Appraisal syntax (rails-3 instead of rails3)

### CODE REVIEW

**Comment by compwron** [comment]
I think I got a little carried away. I don't actually know what I'm doing.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+log
```
Comment (grosser) [inline .gitignore]: good to have that on your global .gitignore just to be save, but no harm in adding it here ...

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
+sudo: false
 rvm:
-  - ree
-  - 1.9.2
-  - 1.9.3
```
Comment (grosser) [inline .travis.yml]: 2.0.0 is nit dead yet, so I'd try to support it if it does not break a bunch ...

File: ar_after_transaction.gemspec [hunk 3]
```diff
@@ -8,5 +8,7 @@ Gem::Specification.new "ar_after_transaction", ARAfterTransaction::VERSION do |s
   s.homepage = "http://github.com/grosser/ar_after_transaction"
   s.files = `git ls-files`.split("\n")
   s.add_runtime_dependency "activerecord"
+  s.add_runtime_dependency "rack"
```
Comment (grosser) [inline ar_after_transaction.gemspec]: it needs rack ... that sounds wrong ...

File: spec/after_initialize_spec.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
 require 'active_record'
+require'rails'
```
Comment (grosser) [inline spec/after_initialize_spec.rb]: why do we need all of rails ?

**Comment by grosser** [comment]
that looks pretty good, can we kick out railties + rack again ?

File: spec/after_initialize_spec.rb (hunk 4 above)
Comment (grosser) [inline spec/after_initialize_spec.rb]: ok ... got it ... we mess with initializers in the test :/

**Comment by grosser** [comment]
doing a manual merge atm, plz don't push any changes

**Comment by grosser** [comment]
merged via https://github.com/grosser/ar_after_transaction/pull/12

**Comment by grosser** [comment]
thx, all up tp date now :)
do you need a new gem release ?

**Comment by compwron** [comment]
Awesome! Many thanks. A new gem release would be great. :)

**Comment by grosser** [comment]
hmm looks like there is no new lib or gemspec code ... new release would not contain anything new ?

**Comment by compwron** [comment]
Good point, never mind.

**Pull request closed without merging**